gomodcmd.go

```diff
@@ -56,29 +56,34 @@ func listModules(modules ...string) (mods map[string]*listModule, err error) {
 	return mods, nil
 }
 
-func goModInfo() (*modfile.File, error) {
+func goModInfo() (*listModule, *modfile.File, error) {
+	if mainMod != nil && mainModFile != nil {
```

Comment:
> I'm not keen on making this a function with side-effects.

Yes, per my Slack message, this was one of the tidy-ups I suggested. `help` is only situation where we should probably not require the user to be in a module context.

> Also, I wonder if it might be slightly nicer just to continue to return *modfile.File

Agreed. I'll do that and see what it looks like.